server/app/services/geo/esri/EsriClient.java

```diff
@@ -192,19 +209,21 @@ public CompletionStage<Optional<AddressSuggestionGroup>> getAddressSuggestions(A
               for (JsonNode candidateJson : json.get("candidates")) {
                 JsonNode location = candidateJson.get("location");
                 JsonNode attributes = candidateJson.get("attributes");
+
                 AddressLocation addressLocation =
                     AddressLocation.builder()
                         .setLongitude(location.get("x").asDouble())
                         .setLatitude(location.get("y").asDouble())
                         .setWellKnownId(wkid)
                         .build();
+
                 Address candidateAddress =
                     Address.builder()
-                        .setStreet(attributes.get("Address").asText())
-                        .setLine2(attributes.get("SubAddr").asText())
-                        .setCity(attributes.get("City").asText())
-                        .setState(attributes.get("RegionAbbr").asText())
-                        .setZip(attributes.get("Postal").asText())
+                        .setStreet(candidateJson.get("address").asText())
+                        .setLine2(attributes.get("SubAddr") == null ? address.getLine2() : attributes.get("SubAddr").asText())
+                        .setCity(attributes.get("City") == null || attributes.get("City").isEmpty() ? address.getCity() : attributes.get("City").asText())
+                        .setState(attributes.get("RegionAbbr") == null ? address.getState() : attributes.get("RegionAbbr").asText())
+                        .setZip(attributes.get("Postal") == null ? address.getZip() : attributes.get("Postal").asText())
```

Comment:
Should all these null checks also have isEmpty checks? I recommend using Guava's com.google.common.base.Strings.isNullOrEmpty.  https://guava.dev/releases/21.0/api/docs/com/google/common/base/Strings.html#isNullOrEmpty-java.lang.String-

Reply:
Unfortunately we can't cast a JsonNode to a string to use that without doing the null check first. I did add the empty checks to the other items as well.